record_test.go

```diff
@@ -71,3 +72,46 @@ func TestRecord_EncodeDecode(t *testing.T) {
 		{ref: 13, intervals: Intervals{{Mint: 5000, Maxt: 1000}}},
 	}, decTstones)
 }
+
+// TestRecord_Corruputed ensures that corrupted records return the correct error.
+// Bugfix check for pull/521 and pull/523.
```

Comment:
This doesn't exactly test pull/521 though the code is the same. 

[Not related to this PR] On that point, why not merge `index/encoding_helpers.go` and `encoding_helpers.go`. Would it make sense to move it to `tsdbutil`? This can be done in another PR. /cc @krasi-georgiev

Reply:
> [Not related to this PR] On that point, why not merge `index/encoding_helpers.go` and `encoding_helpers.go`. Would it make sense to move it to `tsdbutil`? This can be done in another PR. /cc @krasi-georgiev

yes a good idea. @naivewong do you want to try and move both duplicates in tsdbutil? Do it in a separate PR and will merge that one first.

this way the same test will cover both bugs.